lib/cli/parse.js

```diff
@@ -39,6 +37,12 @@ const parser = require('yargs')('run --help')
       type: 'array',
     },
   })
+  .coerce('config', configPath => {
+    /* istanbul ignore next */
+    return require.resolve(configPath, {
+      paths: [process.cwd()],
+    });
+  })
```

Comment:
Use `require.resolve` instead of `path.resolve` to support configs in node_modules.

Reply:
Are there any drawbacks for this approach that would make it not backward compatible?

Reply:
After running a few tests, it appears that this will not work for resolving local config files. For example, in this repository, just trying to load the `package.json` file using the `require.resolve` logic above results in an error:

```js
require.resolve("package.json", {
  paths: [process.cwd()],
});
// Thrown: Error: Cannot find module 'package.json'
```

Reply:
@camacho I have tried running it with `node lib/cli/index.js package.json` and did not get an error. Can you provide additional info?

Reply:
Ah, I think what we are seeing is a change in behavior. The previous implementation did not require a `./` notation for local config paths, allowing a user to write `format-package -w --config format-package-config.js`. This change breaks that previous behavior by requiring any config path to starts with ".".

Other configurations for CLIs, as I remember, do not require the config argument to start with a "./" for local files (such as Jest/ESLint/etc). I do worry that we are going against what I would first expect as a new users (although that's just from my perspective). 

Currently, it is possible to create a config file that imports a node module and re-exports to use shared logic, without this change. But supporting the node module directly from the command line is a nice touch as well. Is there a smart way we can support both, with priority given to finding a local file and falling back to looking for a dependency? 

We could also always look for a config file (which yargs makes pretty easy) and allow for an `extends` property or something of the sort like ESLint/TS/Babel/etc.?

I don't know what the right answer is, but I think continuing to support relative config paths that don't start with "./" is meaningful.

Reply:
We could also adopt something like [`cosmiconfig`](https://github.com/davidtheclark/cosmiconfig) so a config file is always looked for automatically and the config file could then export a shared config dependency (like ESLint/Typescript/Babel/etc.). 

This would leave the CLI `-c` option as a relative path to a different config file location.

Reply:
@camacho I like the idea of using `cosmiconfig`. It's highly configurable and I can move the support for using a `node_module` to that. Will update the PR when it is done and all tests are passing. 

When `--config my-format.js` is provided with a file path that exists.

```
[ '/my-project/my-format.js',
  'format-package.js',
  'format-package.yaml',
  'format-package.yml',
  'format-package.json',
  'format-package.config.js',
  'format-package.config.yaml',
  'format-package.config.yml',
  'package.json' ]
```

When `--config my-format` is provided with a module name that exists.

```
[ '/my-project/node_modules/my-format/index.js',
  'format-package.js',
  'format-package.yaml',
  'format-package.yml',
  'format-package.json',
  'format-package.config.js',
  'format-package.config.yaml',
  'format-package.config.yml',
  'package.json' ]
```

Else search for one of these.

```
[ 'format-package.js',
  'format-package.yaml',
  'format-package.yml',
  'format-package.json',
  'format-package.config.js',
  'format-package.config.yaml',
  'format-package.config.yml',
  'package.json' ]
```

If no config is found, return `lib/defaults/index.js.

Reply:
In regards to the extends, I'll make that as a stretch or another PR.

Reply:
@camacho `cosmiconfig` added.

Reply:
@peterkc this is feels like a great approach to finding the config and feels intuitive 👍